src/modules/project-config.ts

```diff
@@ -192,29 +200,69 @@ export class ProjectConfig {
      * all components are used by default.
      *
      * @param onlyUsed Only include components used by the project. Default: true.
+     * @param onlyInstalled Only include components from packages which are installed. Default: false.
      * @returns A list of all components used by the project.
      */
-    getComponents(onlyUsed: boolean = true): ComponentContext[] {
+    getComponents(onlyUsed: boolean = true, onlyInstalled: boolean = false): ComponentContext[] {
+        let componentContexts: ComponentContext[] = [];
+
+        let packageConfigs = this._packages;
+        if (onlyInstalled) {
+            packageConfigs = packageConfigs.filter((pkg) => pkg.isPackageInstalled());
+        }
+
+        for (const packageConfig of packageConfigs) {
+            const components = this.getComponentsForPackageConfig(packageConfig, onlyUsed);
+            componentContexts = componentContexts.concat(components);
+        }
+
+        return componentContexts;
+    }
+
+    /**
+     * Return all components from installed packages.
+     *
+     * @param onlyUsed Only include components used by the project. Default: true.
+     * @returns A list of components from all installed packages.
+     */
+    getComponentsFromInstalledPackages(onlyUsed: boolean = true): ComponentContext[] {
```

Comment:
is this method really necessary? i think what you would like to achieve is also achievable with the previous getComponents() please try to avoid introducing this method

Reply:
Removed the method and used getComponents(undefined, true) instead to get all components which are from already installed packages.